src/process.cr

```diff
@@ -50,15 +50,18 @@ class Process
     Crystal::System::Process.signal(pid, signal.value)
   end
 
+  alias Interrupt = Crystal::System::Process::Interrupt
```

Comment:
There shouldn't be any `Crystal::System` types in the public API; this type should be defined directly under `Process`, say `InterruptType`.

Alternatively I would suggest defining `TerminalDisconnected` and `SessionEnded` under `Process::ExitReason` and then using that enum. (This would overload the enum's intended usage, but I think that is fine if we interpret this as a "human-caused" exit reason.)